app/konnect/gateway-manager/plugins/update-custom-plugin.md

```diff
@@ -0,0 +1,142 @@
+
+  In this situation, you only need to make sure that each data plane node has the correct logic. 
+  The schema on the control plane doesn't need to be updated.
+
+* **Changes to plugin schema:** If the schema needs to change, you must update both the schema in 
+{{site.konnect_short_name}} and the plugin code itself on each data plane node.
+
+There is no versioning for custom plugins. 
+If you need to version a schema (that is, maintain two or more similar copies of a custom plugin), upload it as a new custom plugin with a unique name. 
+For example, if your original plugin is named `delay`, you would have to name the copy 
+something else, such as `delay-v2`.
+
+## Updating a custom plugin
+
+### How the {{site.konnect_short_name}} platform reads configuration
+
+When a schema is updated in {{site.konnect_short_name}}, the {{site.konnect_short_name}} 
+platform doesn't trigger payload reconciliation automatically.
+This means that if you don't make any configuration changes, such as adding, 
+modifying, or deleting a {{site.base_gateway}} entity, the data plane nodes won't receive a
+payload update.
+
+When pushing changes to the control plane, the payload reconciliation only affects 
+data plane nodes if an instance of the plugin that uses the updated schema has its configuration 
+changed.
+
+Since plugin configurations are stored as JSON blobs, a schema change alone doesn't impact the 
+plugin configuration. However, if the plugin itself is updated, the new schema affects how 
+the new plugin configuration is represented.
+
+### Custom plugin update path
+
+When you need to make plugin changes, we recommend updating the schema in 
+{{site.konnect_short_name}} first, and then on the data plane nodes:
+
+1. Start a migration window.
+1. Update plugin schema in {{site.konnect_short_name}}.  
+1. (Optional) Update configuration for existing plugin instances.
+
+    Based on the nature of the schema updates, you might need this optional step 
+    after updating the schema in {{site.konnect_short_name}} to make sure that any
+    existing plugin entities are updated before the schemas are 
+    changed in the data plane nodes.
+
+1. Update plugin schema on each data plane node.
+1. Stop the migration window.
+
+{:.important}
+> **Important**: In cases where a breaking change is made to the schema, we **do not** 
+recommend updating data plane nodes first. 
+The change will affect each data plane node as soon as the node receives its first payload. 
+This will happen even if there are no changes in the control plane for existing or new plugins 
+using the updated schema. 
+
+See the following table for a comparison of possible changes and upgrade paths, in the case of 
+a configuration parameter change in a custom plugin's schema:
+
+| | Required Default | Required Non-default | Non-required Default | Non-required Non-default |
+|--|--|--|--|--|
+| Add | ✅ | ⏸️ | ✅ | ✅ |
+| Remove | ⏸️ | ❌ | ⏸️ | ⏸️ |
+| Type Change | ❌ | ❌ | ❌ | ❌ |
```

Comment:
The icons on this chart are throwing me off. Instead of ⏸️ maybe we could use ⚠️? 


I do wonder if there is a better way to express this concept without the icons, maybe with words?